pkg/genericapiserver/genericapiserver.go

```diff
@@ -128,6 +131,18 @@ type APIGroupVersionOverride struct {
 	ResourceOverrides map[string]bool
 }
 
+// Info about a group version.
+type GroupVersionInfo struct {
+	GroupMeta latest.GroupMeta
```

Comment:
IIUC, every group/version will have a GruopVersionInfo dedicated to it, and GroupVersionInfo.GroupMeta.Codec should be for the codec that version, but because GroupMeta is populated from latest.Group(<group>), so the Codec will always be the one for the latest version. I see the old code also has this problem.

I think we should define another struct and populate some fields from latest.GroupMeta, like the RESTMapper field, and we don't need GroupMeta.GroupVersions and GroupMeta.InterfacesFor.